readme.md

```diff
@@ -13,6 +14,203 @@ make all
+
+    ...
+
+    # Allow deploy user to manipulate jobs
+    deployuser        ALL=(deployuser) NOPASSWD: ALL, (root) NOPASSWD: UPSTART, UPEXPORT
+
+
+#### Usage
+
+Gem is able to process two versions of Procfiles, format of the Procfile is
+defined in the `version` key. If the key is not present or is not equal to `2`
+file will be parsed as Procfile v.1.
+
+##### Procfile v.1
+
+After upstart-exporter is installed and configured, you may export background jobs
+from an arbitrary Procfile-like file of the following format:
+
+```yaml
+    cmdlabel1: cmd1
+    cmdlabel2: cmd2
+```
+
+i.e. a file `./myprocfile` containing:
+
+```yaml
+    my_tail_cmd: /usr/bin/tail -F /var/log/messages
+    my_another_tail_cmd: /usr/bin/tail -F /var/log/messages
+```
+
+For security purposes, command labels are allowed to contain only letters, digits, and underscores.
+
+##### Procfile v.2
+
+Another format of Procfile scripts is YAML config. A configuration script may
+look like this:
+
+```yaml
+    version: 2
+    start_on_runlevel: 3
+    stop_on_runlevel: 3
+    env:
+      RAILS_ENV: production
+      TEST: true
+    working_directory: /srv/projects/my_website/current
+    commands:
+      my_tail_cmd:
+        command: /usr/bin/tail -F /var/log/messages
+        respawn:
+          count: 5
+          interval: 10
+        env:
+          RAILS_ENV: staging # if needs to be redefined or extended
+        working_directory: '/var/...' # if needs to be redefined
+      my_another_tail_cmd:
+        command: /usr/bin/tail -F /var/log/messages
+        kill_timeout: 60
+        respawn: false # by default respawn option is enabled
+      my_one_another_tail_cmd:
+        command: /usr/bin/tail -F /var/log/messages
+        log: /var/log/messages_copy
+      my_multi_tail_cmd:
+        command: /usr/bin/tail -F /var/log/messages
+        count: 2
+```
+
+`start_on_runlevel` and `stop_on_runlevel` are two global options that can't be
+redefined per-command.
+
+`working_directory` will generate the following line:
+
+    cd 'your/working/directory' && your_command
+
+`env` params can be redefined and extended in per-command options. Note that
+you can't remove a globally defined `env` variable.
+For Procfile example given earlier the generated command will look like:
+
+    env RAILS_ENV=staging TEST=true your_command
+
+`log` option lets you override the default log location (`/var/log/fb-my_website/my_one_another_tail_cmd.log`).
+
+`kill_timeout` option lets you override the default process kill timeout of 30 seconds.
+
+`respawn` option controls how often the job can fail. If the job restarts more
+often than `count` times in `interval`, it won't be restarted anymore.
+
+Options `working_directory`, `env`, `log`, `respawn` can be
+defined both as global and as per-command options.
+
+#### Exporting
+
+To export a Procfile you should run
+
+    sudo upstart-export -p ./myprocfile -n myapp -f format
```

Comment:
init-exporter not require `-n` argument. Currently command look like `sudo upstart-export -p ./myprocfile -f format myapp`. `-n` still supported only for compatibility with upstart-exporter.